enhancements/microshift/optionally-deploy-csi-plugin.md

```diff
@@ -0,0 +1,209 @@
+  without the components.
+
+### Non-Goals
+
+- Generalizing this design to install any other CSI driver.
+- Uninstalling the CSI driver or snapshotter rpms will not uninstall the cluster level components. This is a data safety
+  concern that should be handled by the user.
+- Support installing CSI snapshotter, but not the CSI driver. The snapshotter depends on the CSI driver; it cannot
+  run as a standalone component.
+
+## Proposal
+
+Extract away deployment of the LVMS CSI driver and LVMS CSI snapshot controller from MicroShift logic and provide the
+components as separate, optional rpms. Users should be able to install these rpm simultaneously with core MicroShift
+rpms. It should be possible to install the CSI components to a cluster that was initially deployed without them.
+This should not be a reversible process: uninstalling the rpms should not delete CSI components or cluster APIs from the
+cluster. Doing so endangers users' data and could lead to orphaning of LVM volumes.
+
+### Workflow Description
+
+**_Prerequisites_**
+
+- Host has LVM installed
+
+**_Installation with CSI Driver and Snapshotting_**
+
+1. User determines there is a requirement for persistent storage and volume snapshotting.
+2. User specifies an ostree blueprint which includes the following sections:
+    1. Packages: microshift, microshift-greenboot, microshift-networking, microshift-selinux, microshift-lvms, 
+       microshift-lvms-snapshotting
+    2. File (Optional): lvmd.yaml
+3. User compiles an ostree commit from the blueprint
+4. User deploys the ostree commit to host
+5. MicroShift host boots
+6. MicroShift starts
+7. MicroShift deploys:
+   1. LVMS CSI manifests
+   2. LVMS Snapshot manifests
+
+**_Installation without CSI Driver and Snapshotting_**
+
+1. User determines there is not a requirement for persistent storage and volume snapshotting.
+2. User specifies an ostree blueprint which includes the following sections:
+    1. Packages: microshift, microshift-greenboot, microshift-networking, microshift-selinux
+3. User compiles an ostree commit from the blueprint
+4. User deploys the ostree commit to host
+5. MicroShift host boots
+6. MicroShift starts
+
+**_Day-1|2 Installation_**
+
+1. User has previous installed MicroShift on a device.
+2. User later determines there is a requirement for persistent storage and volume snapshotting.
+3. User specifies an ostree blueprint which includes the following sections:
+    1. Packages:  microshift-lvms, microshift-lvms-snapshotting
+    2. File (Optional): lvmd.yaml
+4. User compiles an ostree commit from the blueprint
+5. User deploys the ostree commit to host
+6. MicroShift host reboots
+7. MicroShift starts
+8. MicroShift deploys:
+   1. LVMS CSI manifests
+   2. LVMS Snapshot manifests
+
+### API Extensions
+
+- None
+
+### Topology Considerations
+
+#### Single-node Deployments or MicroShift
+
+The changes proposed here only affect MicroShift.
```

Comment:
They also affect LVMS due Microshift and LVMS sharing a version dependency to each other. We will no longer need to release LVMS ahead of Microshift technically and can version it separately. That also means the RPMs need to be versioned and released separately. Otherwise I believe there is no gain here for extracting them

Reply:
The gain is also to support another storage provider in the future.
We need to discuss how this change affects LVMS release cycle.

Reply:
Looking at the rebase changes, we assume that the RPM is still built in the MicroShift repo, so we maintain the version dependency. In general, change the build procedure and full decoupling should be treated as a separate ER, I think.

Reply:
Agreed, I think that the first step should remain scoped to the feature work.